### PR TITLE
Use GitHub Actions instead of Travis CI

### PR DESCRIPTION
**Pull request by ind1go**
Moves to a workflow, analogous 🤞  with the previous Travis CI configuration.

Note: currently uses @bjhargrave's fix to actions/setup-java#200 to deal with GPG subkeys. Revert to `actions/setup-java@v2` once that is merged.

### CODE REVIEW

File: .github/workflows/maven-build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  build:
+    name: Maven Build
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: bjhargrave/setup-java@a1600c1f88a7741e2420b0a657f2838cec4f09c5
```
Comment (ind1go) [inline .github/workflows/maven-build.yml:14]: When upstream action is fixed, change this to:

```
uses: actions/setup-java@v2
```

**Comment by stewartfrancis** [comment]
I notice we're no longer using the settings.xml which sets the properties `gpg.keyname` and `gpg.passphrase`, which seem like default variable names used by the `maven-gpg-plugin`.  Are they getting set by some other mechanism now?  I couldn't find any reference to the gpg plugin using environment variables in its documentation

**Comment by ind1go** [comment]
> I notice we're no longer using the settings.xml which sets the properties `gpg.keyname` and `gpg.passphrase`, which seem like default variable names used by the `maven-gpg-plugin`. Are they getting set by some other mechanism now? I couldn't find any reference to the gpg plugin using environment variables in its documentation

That's right, the `settings.xml` is constructed by the `setup-java` action. You can find more info on the GPG bit in particular [here](https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Publishing-using-Apache-Maven).

**Comment by stewartfrancis** [comment]
Well that's quite nice isn't it!  lgtm!

**Comment by stewartfrancis** [comment]
To be clear - I think we should go ahead and merge this, switching this repo over to GH actions

**Pull request merged**